### PR TITLE
Using gitlab as a container registry

### PR DESCRIPTION
**Pull request by VariableDeclared**
Tutorial on using gitlab as a Container Registry

### CODE REVIEW

File: tutorials/containers/using-gitlab-as-a-container-registry/using-gitlab-as-a-container-registry.md [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+author: Peter De Sousa <peter.de.sousa@canonical.com>
+---
+
+# Using GitLab as a container registry
+
+## Introduction
+
+The aim of this tutorial post is to demonstrate how you can use Charmed Kubernetes in a CI/CD environment using GitLab’s container registry functionality.
+
+GitLab has the ability to store upto 10GB in a container registry for projects, you can incorporate the building of these containers into your own CI/CD pipeline or you can use Gitlab’s own CI/CD functionality to do this for you. For the purposes of this blog post you will do this by hand so you can get a grasp of the process.
```
Comment (tytus-kurek) [inline tutorials/containers/using-gitlab-as-a-container-registry/using-gitlab-as-a-container-registry.md]: Please expand a little bit more about why we're doing that. Tell the story. What is the problem? How are you trying to solve it?

File: tutorials/containers/using-gitlab-as-a-container-registry/using-gitlab-as-a-container-registry.md [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+feedback_url: https://github.com/canonical-websites/tutorials.ubuntu.com/issues
+published: 2019-09-02
+author: Peter De Sousa <peter.de.sousa@canonical.com>
+---
+
+# Using GitLab as a container registry
+
+## Introduction
+
+The aim of this tutorial post is to demonstrate how you can use Charmed Kubernetes in a CI/CD environment using GitLab’s container registry functionality.
```
Comment (tytus-kurek) [inline tutorials/containers/using-gitlab-as-a-container-registry/using-gitlab-as-a-container-registry.md]: Move it below the next paragraph and structure in a form of bullet points (you can reference some recent tutorials like https://tutorials.ubuntu.com/tutorial/install-microk8s-on-mac-os#0)

File: tutorials/containers/using-gitlab-as-a-container-registry/using-gitlab-as-a-container-registry.md [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+difficulty: 3
+status: draft
+feedback_url: https://github.com/canonical-websites/tutorials.ubuntu.com/issues
+published: 2019-09-02
+author: Peter De Sousa <peter.de.sousa@canonical.com>
+---
+
+# Using GitLab as a container registry
+
+## Introduction
```
Comment (tytus-kurek) [inline tutorials/containers/using-gitlab-as-a-container-registry/using-gitlab-as-a-container-registry.md]: Overview

File: tutorials/containers/using-gitlab-as-a-container-registry/using-gitlab-as-a-container-registry.md [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+categories: containers
+tags: kubernetes, containers, gitlab, cicd
+difficulty: 3
+status: draft
+feedback_url: https://github.com/canonical-websites/tutorials.ubuntu.com/issues
+published: 2019-09-02
+author: Peter De Sousa <peter.de.sousa@canonical.com>
+---
+
+# Using GitLab as a container registry
```
Comment (tytus-kurek) [inline tutorials/containers/using-gitlab-as-a-container-registry/using-gitlab-as-a-container-registry.md]: for Charmed Kubernetes

File: tutorials/containers/using-gitlab-as-a-container-registry/using-gitlab-as-a-container-registry.md [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+
+## Introduction
+
+The aim of this tutorial post is to demonstrate how you can use Charmed Kubernetes in a CI/CD environment using GitLab’s container registry functionality.
+
+GitLab has the ability to store upto 10GB in a container registry for projects, you can incorporate the building of these containers into your own CI/CD pipeline or you can use Gitlab’s own CI/CD functionality to do this for you. For the purposes of this blog post you will do this by hand so you can get a grasp of the process.
+
+## Before you start
```
Comment (tytus-kurek) [inline tutorials/containers/using-gitlab-as-a-container-registry/using-gitlab-as-a-container-registry.md]: Make it a sub-section of the "Overview" section ("You will only need ..."). Again, you can reference some of the latest tutorials.

File: tutorials/containers/using-gitlab-as-a-container-registry/using-gitlab-as-a-container-registry.md [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+
+Find yourself a healthy kubernetes cluster, or if you don’t have access to a cluster, head over to [MicroK8s](https://tutorials.ubuntu.com/tutorial/install-a-local-kubernetes-with-microk8s#0) and install one on your laptop at no cost to yourself, if you’re on Windows you can follow [this guide](https://itnext.io/microk8s-on-windows-the-canonical-way-ed15fd4e5476).
+
+![Charmed Kubernetes](./images/charmed_kubernetes.png)
+
+## Process
+To create our container registry on gitlab you will need to complete the following steps:
```
Comment (tytus-kurek) [inline tutorials/containers/using-gitlab-as-a-container-registry/using-gitlab-as-a-container-registry.md]: your

File: tutorials/containers/using-gitlab-as-a-container-registry/using-gitlab-as-a-container-registry.md [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+
+- Create a project
```
Comment (tytus-kurek) [inline tutorials/containers/using-gitlab-as-a-container-registry/using-gitlab-as-a-container-registry.md]: Could you please add at least a couple of words about each of those bullet points? This page looks a little empty at the moment.

File: tutorials/containers/using-gitlab-as-a-container-registry/using-gitlab-as-a-container-registry.md [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+       securityContext: {}
+      terminationGracePeriodSeconds: 30
+```
+OK lets check the logs now:
+
+![Program is logging](./images/logging_screenshot.png)
+
+Voila!
+
+# Conclusion
```
Comment (tytus-kurek) [inline tutorials/containers/using-gitlab-as-a-container-registry/using-gitlab-as-a-container-registry.md]: Please make it a separate section ("That's all folks!"). Again, you can reference one of the latest tutorials. Also, add some interesting links to our products to let the reader continue their story.

File: tutorials/containers/using-gitlab-as-a-container-registry/using-gitlab-as-a-container-registry.md (hunk 3 above)
Comment (tytus-kurek) [inline tutorials/containers/using-gitlab-as-a-container-registry/using-gitlab-as-a-container-registry.md]: Add "Duration" everywhere.

**Comment by VariableDeclared** [comment]
@tytus-kurek Made those changes :smile:

**Comment by tytus-kurek** [comment]
@caldav ready for review

**Pull request merged**